src/Fl_Input.cxx

```diff
@@ -278,7 +278,7 @@ int Fl_Input::kf_undo() {
 // Redo. (currently unimplemented.. toggles undo() instead)
 int Fl_Input::kf_redo() {
   if (readonly()) { fl_beep(); return 1; }
-  return kf_undo();                     // currently we don't support multilevel undo
+  return redo();                     // currently we don't support multilevel undo
 }
```

Comment:
The comment(s) above this "currently unimplemented" and in the line above "currently we don't support multilevel undo" appear to be outdated. Isn't this now implemented? I see that you changed kf_**undo** to **redo** but this looks like it was intended. Only the comments are questionable.